### ISSUE TITLE
Bank HaPoalim is asking for 2FA in chromium

### ISSUE THREAD

**Issue by liranelisha** [eshaham/israeli-bank-scrapers#823]
starting last week, Bank HaPoalim is asking for elevated privileges via 2FA through the phone for unknown sessions, causing each run from the scraper to prompt the 2FA

**Comment by daniel-hauser** [comment]
Where do you run the scraper from? It might be related to the location of the servers.
I'm running in GitHub actions and never got a 2fa request

**Comment by liranelisha** [comment]
i run it locally from a machine in israel

**Comment by JavierChames** [comment]
Hi ,i have the same issue(running the code on raspberry pi)-any  idea for a solution?

**Comment by baruchiro** [comment]
@galbarm

**Comment by gbarmashiahflir** [comment]
Happens to me to well.
What's weird is that when testing with `showBrowser: true`, the login finishes successfully and no 2FA is requested.

**Comment by liranelisha** [comment]
@gbarmashiahflir yeah, that's my mitigation as well. when scrapping hapoalim i show browser. that makes sense as the headlesss browser and headed browser has different signature, making the headed browser show the 2fa once, and once solved not showing it again. however this isn't possible with headless browser.
waiting to see if there could be a more reliable solution

**Comment by ranrib** [comment]
I'm experiencing the same issue. It looks like when I'm trying a new machine/IP (changing somehow my signature), it asks for 2FA again.

**Comment by JavierChames** [comment]
i found a  "kind" of workaround for my case, I use raspberry pi, usually i installed a light version of Raspbian os(OS for raspberry pi) with no desktop(only cli)-this time i installed the desktop version enter to the browser-get the SMS to enter the code, enter to the site. and then tried again the script-this time it worked..not the best-but for my case its seems to work 🤣

**Comment by JavierChames** [comment]
Hi
the issues occurs again..it worked and somehow it request for 2F now.any idea anyone?

**Comment by liranelisha** [comment]
it reoccurs on my machine as well every 3 or 4 months

**Comment by JavierChames** [comment]
@liranelisha any idea for solution?

**Comment by JavierChames** [comment]
@sebikaplun maybe you? :-) regarding the issue of 2FA for Bank hapoalim?

**Comment by sr01** [comment]
An OTP is required when using a new device. The device identifier is stored in local storage. To persist it, Puppeteer must be configured with userDataDir.

**Comment by JavierChames** [comment]
The issue is that its not a new device...
For example, i run it from Raspberry pi, when i open the browser there is no issue.
But if i run the script it requires the 2FA...

**Comment by itaigrady** [comment]
any workaround for this issue?

**Comment by liranelisha** [comment]
@itaigrady don't know if a real W/A, but i start this one in `showBrowser: true` mode every time it fails on password related issue, so i can see if it asks for 2FA again

**Comment by tomerh2001** [comment]
> @gbarmashiahflir yeah, that's my mitigation as well. when scrapping hapoalim i show browser. that makes sense as the headlesss browser and headed browser has different signature, making the headed browser show the 2fa once, and once solved not showing it again. however this isn't possible with headless browser.
> waiting to see if there could be a more reliable solution

It works, but I need it to run it inside a docker container? I don't mind doing the show browser once in a while, but how do I pass its "effects" (cache I assume) into my docker container?

**Comment by daniel-hauser** [comment]
> It works, but I need it to run it inside a docker container? I don't mind doing the show browser once in a while, but how do I pass its "effects" (cache I assume) into my docker container?

You can mount a folder to the container, then pass it as `--user-data-dir` in the chrome args.

I'm really interested in the reason they ask for MFA, i have a headless scraper that runs twice a day in a new isolated container for years and i never got to experience this issue.
